enhancements/api-review/new-CRD-ManagedClusterVersion.md

```diff
@@ -0,0 +1,452 @@
+---
+title: managed-cluster-version-crd
+authors:
+  - @2uasimojo
+reviewers: # Include a comment about what domain expertise a reviewer is expected to bring and what area of the enhancement you expect them to focus on. For example: - "@networkguru, for networking aspects, please look at IP bootstrapping aspect"
+  - @wking
+  - @jewzaam
+  - @cblecker
+approvers: # A single approver is preferred, the role of the approver is to raise important questions, help ensure the enhancement receives reviews from all applicable areas/SMEs, and determine when consensus is achieved such that the EP can move forward to implementation.  Having multiple approvers makes it difficult to determine who is responsible for the actual approval.
+  - @LalatenduMohanty
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - @deads2k
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com//browse/HIVE-2366
+see-also:
+  - n/a
+replaces:
+  - n/a
+superseded-by:
+  - n/a
+---
+
+# ManagedClusterVersion CRD
+
+## Summary
+
+Introduce a new namespaced openshift/api CustomResourceDefinition,
+`ManagedClusterVersion`, as a vehicle for managers of OpenShift fleets to
+expose version and upgrade information in the management cluster.
+
+## Motivation
+
+OpenShift fleet management software is stratified. For example, ACM (Advanced
+Cluster Manager) and OCM (OpenShift Cluster Manager, the front end for the
+Managed OpenShift product) both provide users access and visibility into
+clusters which may internally be controlled by different pieces of fleet
+management software such as [hive](https://github.com/openshift/hive/) and
+[hypershift](https://github.com/openshift/hypershift/). In this picture, hive
+and hypershift interact directly with the "spoke" clusters, whereas ACM and OCM
+rely on data visible only on the "hub" (the management cluster on which the
+hive/hypershift/ACM/OCM software is running). Thus, any spoke-specific
+information \*CM requires to operate must be brokered by the hive/hypershift
+layer.
+
+These different levels may have different responsibilities. Salient to this
+enhancement, the \*CM layer is responsible for driving upgrades. To do so,
+it needs visibility into the spoke cluster's ClusterVersion data. Today the
+only mechanisms available for accessing this information entail logging into,
+or running an agent in, the spoke cluster. This is not ideal:
```

Comment:
What is the example of the agent? Klusterlet (https://operatorhub.io/operator/klusterlet) ?

Reply:
I think so, yes.

Except does klusterlet have a way to initiate communication with the hub, or does it only _pull_ from the hub?

I get confused with the different OCMs -- is this the one ACM uses? Does Hypershift use this one as well?

In any case, is it worth mentioning/discussing in the document?